subprojects/logging/src/main/java/org/gradle/internal/deprecation/DeprecationMessageBuilder.java

```diff
@@ -53,6 +53,26 @@ public DeprecationMessageBuilder guidedBy(String documentationId, String section
         return this;
     }
 
+    public DeprecationMessageBuilder guidedBy(String documentationId) {
+        this.documentationReference = DocumentationReference.create(documentationId);
+        return this;
+    }
+
+    public DeprecationMessageBuilder withDslReferenceForType(Class<?> targetClass) {
```

Comment:
For now, I think we should avoid linking to the entire DSL reference for a type or Javadoc and only specifically link to 1) upgrade guide section, 2) user manual or 3) specific DSL reference property.  Everything else is too broad I think.